### DISCUSSION TITLE
Problem in visual node hierarchy

### DISCUSSION THREAD

**Discussion by AlexAS81** [OP-TED/eForms-SDK#352] [Q&A] [answered]
We have seen that when using the eForms Editor application (eforms-sdk.1.5.1), there are some nodes whose parentNode does not match in the visual model and the fields.json. Is this correct?

![image](https://user-images.githubusercontent.com/115476802/221189139-0d8f47bd-2fe9-4e32-920e-c29dd2ada9b0.png)

For example 10,json:

![image](https://user-images.githubusercontent.com/115476802/221187491-c4c81410-3bbb-48ae-8b8d-aa54625a54bc.png)

fields.json:

![image](https://user-images.githubusercontent.com/115476802/221189080-17d68510-e9e4-46c0-ba88-81d6002a1be0.png)

Is there any problem in the generation of the models due to this problem that is reported by console?

Thanks in advance
Alex

**Comment by rouschr** [comment, marked as answer]
Hello,

some of these are real problems in the hierarchy (usually in the notice types) and some could be false positives or just more desirable. We progressively fix these problems as we find them, focusing on the real problems first.
I think one of the problems about `"ND-LotsGroup"` and `BT-330-Procedure` has been fixed recently, the SDK 1.6.0 will be released very soon, try again when it comes out.

**Reply by teslavitas** [reply]
@rouschr Hello, 
in the Notice editor, there is a method that tries to fix such situations by adding extra nodes to the conceptual mode.
Should we implement something similar in our solution or expect that by October there will be no collisions?
Notice editor method name is addIntermediaryNonRepeatingNodesRec, in class VisualModel.

**Reply by rouschr** [reply]
@teslavitas I think you should do the same as the editor demo for this case, you can reuse or copy that code.
Various things will get simpler as we add more data / attributes in the fields.json, the editor demo is always catching up to 
fields.json
There will be various data improvements in SDK 1.8 which will help with XML generation.

**Comment by AlexAS81** [comment]
We are using SDK version 1.7 and for example with BT-137-Lot we find that in the notice type its parentNodeId is the ND-LotTenderingProcess and the fields.json indicates that its parentNodeId is the ND-Lot. And ND-Lot being a repeatable field, when generating the conceptual we cannot generate it well. 

For the non-repeatable fields we have the "addIntermediaryNonRepeatingNodesRec" method, but for the repeatable ones we understand that we still need to unify the parentNodeId of the notice types with those of the fields.json. Is there any progress on this?

When will SDK 1.8 be released?

**Comment by rouschr** [comment]
> When will SDK 1.8 be released?

In due time, but soon.
I recommend you can look at the RC for now and see if it solves your problems:
https://github.com/OP-TED/eForms-SDK/tree/1.8.0-rc.1
Many form and xml related problems have been solved in 1.8 and more improvements/fixes coming in 1.9
